osu.Game/Beatmaps/Drawables/BeatmapGroup.cs

```diff
@@ -40,36 +40,41 @@ public class BeatmapGroup : IStateful<BeatmapGroupState>
         public BeatmapSetInfo BeatmapSet;
 
         private BeatmapGroupState state;
+
         public BeatmapGroupState State
         {
             get { return state; }
             set
             {
                 state = value;
-
-                switch (value)
-                {
-                    case BeatmapGroupState.Expanded:
-                        Header.State = PanelSelectedState.Selected;
-                        foreach (BeatmapPanel panel in BeatmapPanels)
-                            panel.State = panel == SelectedPanel ? PanelSelectedState.Selected : PanelSelectedState.NotSelected;
-                        break;
-                    case BeatmapGroupState.Collapsed:
-                        Header.State = PanelSelectedState.NotSelected;
-                        foreach (BeatmapPanel panel in BeatmapPanels)
-                            panel.State = PanelSelectedState.Hidden;
-                        break;
-                    case BeatmapGroupState.Hidden:
-                        Header.State = PanelSelectedState.Hidden;
-                        foreach (BeatmapPanel panel in BeatmapPanels)
-                            panel.State = PanelSelectedState.Hidden;
-                        break;
-                }
-
+                UpdateState();
                 StateChanged?.Invoke(state);
             }
         }
 
+        public void UpdateState()
+        {
+            switch (state)
+            {
+                case BeatmapGroupState.Expanded:
+                    Header.State = PanelSelectedState.Selected;
+                    foreach (BeatmapPanel panel in BeatmapPanels)
+                        panel.State = panel == SelectedPanel ? PanelSelectedState.Selected :
+                            !panel.Filtered ? PanelSelectedState.NotSelected : PanelSelectedState.Hidden;

```

Comment:
I think at the point where you're nesting ternaries, it's probably going too far.

How about:
```
if (panel == SelectedPanel)
  panel.State = PanelSelectedState.Selected;
else if (panel.Filtered)
  panel.State = PanelSelectedState.Hidden;
else
  panel.State = PanelSelectedState.NotSelected;
```